### PR TITLE
List: adopt new live/sticky behaviour

### LINKED ISSUE #353: Adopt live pattern for Lists
### Expected Behavior
We are in the middle of changing a design principle: When do we have "live" objects which generate their values every time, and when do we have sticky objects. 

We shift this to having the existing objects like TextOf from live to sticky. This Right now, TextOf is a live object which builds its values every time.

In this issue, ListOf should be designed like map: ListOf will be List.Of an sticky by default, and List.Live will be the live object needed.

The obsolete objects ListOf and Sticky should be declared with the [Obsolete] attribute.

### Actual Behavior
ListOf is a live object and sticky is the sticky object.

### PR DESCRIPTION
**Pull request by EdJoPaTo**
### Please check if the PR fulfills these requirements
- [x] I made sure that my code builds
- [x] I merged the master into this branch before pushing
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

### What is the current behavior? (You can also link to an open issue here)

close #353

### Does this PR introduce a breaking change? (check one with "x")
- [x] Yes
- [ ] No

### If this PR contains a breaking change, please describe the impact and migration path for existing applications

Namespace `Yaapii.Atoms.List` is renamed to `Yaapii.Atoms.Lists`.

###  Other information

### CODE REVIEW

**Comment by Meerownymous** [comment]
@EdJoPaTo build is failing.

File: src/Yaapii.Atoms/Collection/Sorted.cs [hunk 1]
```diff
@@ -38,7 +38,7 @@ public sealed class Sorted<T> : CollectionEnvelope<T>
         /// A list with default sorting (ascending)
         /// </summary>
         /// <param name="src">the source enumerable</param>
-        public Sorted(params T[] src) : this(new ListOf<T>(src))
+        public Sorted(params T[] src) : this(new List.List.Live<T>(src))
```
Comment (Meerownymous) [inline src/Yaapii.Atoms/Collection/Sorted.cs]: We must not use a namespace which is the same as a class name - this leads to conflicts in our projects.
The current solution is, that we use the plural for the namespace. Texts, Lists, Collections - or another name if applicable (see map area: "Lookup"

File: src/Yaapii.Atoms/List/List.Live.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            { }
+
+            /// <summary>
+            /// ctor
+            /// </summary>
+            /// <param name="src">source enumerable</param>
+            public Live(IEnumerable<T> src) : base(
+                () =>
+                {
+                    var temp = new List<T>();
```
Comment (Meerownymous) [inline src/Yaapii.Atoms/List/List.Live.cs]: As mentioned in another PR - please think about a solution for Live classes which does not copy the Enumerable.

File: src/Yaapii.Atoms/Collection/Sorted.cs (hunk 1 above)
Comment (EdJoPaTo) [reply src/Yaapii.Atoms/Collection/Sorted.cs]: Renaming the namespace results in a breaking change. Should ListOf and Sticky be removed then completly too? Users have to change their code anyway due to the namespace renaming.

Comment (Meerownymous) [reply src/Yaapii.Atoms/Collection/Sorted.cs]: When we have the current change ready, the switch from live to sticky by default will be a big BC also, so we will release Atoms as version 2.0 when we are done. Until then, no new Atoms package will be released.

Comment (Meerownymous) [reply src/Yaapii.Atoms/Collection/Sorted.cs]: I removed Live and Sticky, we should provide an upgrade documentation instead.

**Pull request merged**